hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/common/CachingBlockManager.java

```diff
@@ -330,6 +341,10 @@ private void readBlock(BufferData data, boolean isPrefetch, BufferData.State...
         this.read(buffer, offset, size);
         buffer.flip();
         data.setReady(expectedState);
+
+        if(isPrefetch) {
+          this.prefetchingStatistics.prefetchOperationCompleted();
```

Comment:
what happens if the read failed? is the failure count and duration before failure recorded separately from the successful calls?

Reply:
So this was currently only measuring successful calls, I'm now also adding a count for failure. I can't figure out how to measure duration for a failure though, could you point me to an example?